packages/node_modules/@node-red/editor-client/src/js/ui/contextMenu.js

```diff
@@ -31,7 +31,11 @@ RED.contextMenu = (function () {
             const canEdit = !RED.workspaces.isLocked()
             const canRemoveFromGroup = hasSelection && !!selection.nodes[0].g
             const isAllGroups = hasSelection && selection.nodes.filter(n => n.type !== 'group').length === 0
-            const hasGroup = hasSelection && selection.nodes.filter(n => n.type === 'group' ).length > 0
+            const hasGroup = hasSelection && selection.nodes.filter(n => n.type === 'group').length > 0
+            const hasDisabledNode = hasSelection && selection.nodes.filter(e => e.d).length > 0;
+            const hasEnabledNode = hasSelection && selection.nodes.filter(e => !e.d).length > 0;
+            const hasUnlabeledNode = hasSelection && selection.nodes.filter(e => e.l === false).length > 0;
+            const hasLabeledNode = hasSelection && selection.nodes.filter(e => e.l || e.l === undefined).length > 0;
```

Comment:
This code is now doing lots of separate iterations of `selection.nodes` - and using `filter` means it is creating a new array each time.

It would be better to now have a single `forEach` to iterate over `selection.nodes` and set the flags as necessary.